server/app/models/access_token.rb

```diff
@@ -1,18 +1,202 @@
+require 'uri'
+require_relative '../helpers/config_helper'
+
 class AccessToken
   include Mongoid::Document
   include Mongoid::Timestamps
 
+  include ConfigHelper
+
   belongs_to :user
   validates_presence_of :scopes, :user
 
-  field :token_type, type: String
+  validates :refresh_token, uniqueness: { scope: :internal, allow_nil: true }
+
+  field :token_type, type: String, default: 'bearer'
   field :token, type: String
   field :refresh_token, type: String
-  field :expires_at, type: Time
+  field :expires_at, type: DateTime
   field :scopes, type: Array
+  field :deleted_at, type: BSON::Timestamp, default: nil
+  field :internal, type: Boolean, default: true
```

Comment:
internal: false is for auth provider's tokens. 
internal: true is for authenticating to this master